src/tests/intg/test_secrets.py

```diff
@@ -0,0 +1,143 @@
+#
+# Secrets responder integration tests
+#
+# Copyright (c) 2016 Red Hat, Inc.
+#
+# This is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by
+# the Free Software Foundation; version 2 only
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+import os
+import stat
+import config
+import signal
+import subprocess
+import time
+import pytest
+from util import unindent
+from requests import HTTPError
+
+from secrets import SecretsLocalClient
+
+
+def create_conf_fixture(request, contents):
+    """Generate sssd.conf and add teardown for removing it"""
+    conf = open(config.CONF_PATH, "w")
+    conf.write(contents)
+    conf.close()
+    os.chmod(config.CONF_PATH, stat.S_IRUSR | stat.S_IWUSR)
+    request.addfinalizer(lambda: os.unlink(config.CONF_PATH))
+
+
+def create_sssd_secrets_fixture(request):
+    if subprocess.call(['sssd', "--genconf"]) != 0:
+        raise Exception("failed to regenerate confdb")
+
+    resp_path = os.path.join(config.LIBEXEC_PATH, "sssd", "sssd_secrets")
+
+    secpid = os.fork()
+    if secpid == 0:
+        if subprocess.call([resp_path, "--uid=0", "--gid=0"]) != 0:
+            raise Exception("sssd_secrets failed to start")
+
+    time.sleep(1)
+
+    def sec_teardown():
+        if secpid == 0:
+            return
+
+        os.kill(secpid, signal.SIGTERM)
+        for file in os.listdir(config.SECDB_PATH):
+            os.unlink(config.SECDB_PATH + "/" + file)
+    request.addfinalizer(sec_teardown)
```

Comment:
file is a built-in and that's the reason why it has a different collor in GH.
It's not a big problem that it is redefined but some IDE complains about
such change.